utils/zerovec/src/ule/plain.rs

```diff
@@ -135,6 +135,38 @@ impl AsULE for u8 {
 // EqULE is true because u8 is its own ULE.
 unsafe impl EqULE for u8 {}
 
+// Safety (based on the safety checklist on the ULE trait):
+//  1. [u8; N] does not include any uninitialized or padding bytes.
+//  2. [u8; N] is aligned to 1 byte.
+//  3. The impl of validate_byte_slice() returns an error if any byte is not valid (never).
+//  4. The impl of validate_byte_slice() returns an error if there are leftover bytes.
+//  5. The other ULE methods use the default impl.
+//  6. [u8; N] byte equality is semantic equality
+unsafe impl<const N: usize> ULE for [u8; N] {
```

Comment:
Suggestion/Question: Should we make this impl general over `[U; N]` where `U: ULE`?